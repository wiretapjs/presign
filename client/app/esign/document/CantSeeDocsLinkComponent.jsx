import React, { Component } from 'react'
import autoBind from 'react-autobind'
import * as analytics from './analytics'
import CantSeeDocsLink from './CantSeeDocsLink'

class CantSeeDocsLinkComponent extends Component {
  constructor (props) {
    super(props)
    this.state = {openDialog: false}
    autoBind(this)
  }

  openCantSeeDialog () {
    this.setState({openDialog: true})
    analytics.recordEsignCantSeeDocumentClick()
  }

  closeCantSeeDialog () {
    this.setState({openDialog: false})
  }

  render () {
    return (
      <CantSeeDocsLink
        isOpen={this.state.openDialog}
        openDialog={this.openCantSeeDialog}
        closeDialog={this.closeCantSeeDialog}
      />
    )
  }
}

export default CantSeeDocsLinkComponent
