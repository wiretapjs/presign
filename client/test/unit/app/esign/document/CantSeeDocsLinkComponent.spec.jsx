import React from 'react'
import { shallow } from 'enzyme'
import CantSeeDocsLinkComponent from 'app/esign/document/CantSeeDocsLinkComponent'

describe('CantSeeDocsLinkComponent', () => {
  const CantSeeDocsLinkWrapper = shallow(<CantSeeDocsLinkComponent />)
  const CantSeeDocsLinkInstance = CantSeeDocsLinkWrapper.instance()
  it('Render CantSeeDocsLinkComponent', () => {
    expect(CantSeeDocsLinkWrapper).toMatchSnapshot()
  })
  it('Open Dialog to be true', () => {
    CantSeeDocsLinkInstance.openCantSeeDialog()
    expect(CantSeeDocsLinkWrapper.state().openDialog).toBe(true)
  })
  it('Open Dialog to be false', () => {
    CantSeeDocsLinkInstance.closeCantSeeDialog()
    expect(CantSeeDocsLinkWrapper.state().openDialog).toBe(false)
  })
})
