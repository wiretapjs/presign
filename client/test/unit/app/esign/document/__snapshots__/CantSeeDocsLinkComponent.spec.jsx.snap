// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CantSeeDocsLinkComponent Render CantSeeDocsLinkComponent 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CantSeeDocsLink
    closeDialog={[Function]}
    isOpen={false}
    openDialog={[Function]}
  />,
  "nodes": Array [
    <CantSeeDocsLink
      closeDialog={[Function]}
      isOpen={false}
      openDialog={[Function]}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CantSeeDocsLinkComponent />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CantSeeDocsLinkComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "openDialog": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CantSeeDocsLink
          closeDialog={[Function]}
          isOpen={false}
          openDialog={[Function]}
        />,
        "_debugID": 2,
        "_renderedOutput": <CantSeeDocsLink
          closeDialog={[Function]}
          isOpen={false}
          openDialog={[Function]}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CantSeeDocsLinkComponent />,
}
`;
